app/services/video-encoding-optimizations/definitions.ts

```diff
@@ -0,0 +1,121 @@
+enum EncoderType {
+    x264 = 0,
+    nvenc = 1,
+    amd = 2,
+    qsv = 3,
+}
+
+enum GameType {
+    PUBG = 'PLAYERUNKNOWN\'S BATTLEGROUNDS',
+    LeagueOfLegends = 'League of Legends',
+    Fortnite = 'Fortnite',
+    Destiny2 = 'Destiny 2',
+}
+
+enum PresetType {
+    ultrafast = 'ultrafast',
+    superfast = 'superfast',
+    veryfast = 'veryfast',
+    faster = 'faster',
+    fast = 'fast',
+    medium = 'medium',
+    slow = 'slow',
+    slower = 'slower',
+}
+
+export interface IProfile {
+  profile: string;
+  description: string;
+  preset: PresetType;
+}
+
+const CPU_profile: IProfile = {
+  profile: 'CPU',
+  description: 'CPU optimized (Reduction of the CPU load for an average video quality)',
+  preset: PresetType.veryfast };
+
+const VQ_profile: IProfile = {
+  profile: 'VQ',
+  description: 'Video quality optimized (Improvement of the video quality at a very low CPU usage)',
+  preset: PresetType.ultrafast
+};
+
+export interface IEncoderPreset {
+  profile: IProfile;
+  game: GameType;
+  settings: string;
+  encoder: EncoderType;
+}
+
+export function getPreset() {
+  return Preset;
+}
+
+const Preset: IEncoderPreset[] = [
```

Comment:
I would call this `Presets` (plural).  Also you can just export the variable directly instead of defining a getter:
```
export const Presets: IEncoderPreset[] = [
```

Reply:
Done.